src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberFieldSymbol.cs

```diff
@@ -592,13 +592,23 @@ internal override bool IsDefinedInSourceTree(SyntaxTree tree, TextSpan? definedW
 
         internal override void AfterAddingTypeMembersChecks(ConversionsBase conversions, BindingDiagnosticBag diagnostics)
         {
-            // This check prevents redundant ManagedAddr diagnostics on the underlying pointer field of a fixed-size buffer
-            if (!IsFixedSizeBuffer)
+            var type = Type;
+            if (type.Kind != SymbolKind.PointerType && !IsFixedSizeBuffer)
```

Comment:
>if (type.Kind != SymbolKind.PointerType && !IsFixedSizeBuffer) [](http://example.com/codeflow?start=12&length=62)

It is not obvious why do we still want to perform the check here under this condition?

Reply:
Which check? `IsFixedSizeBuffer`? Given the existing code comment, it shouldn't be necessary here. But my thought was to keep it for two reasons:

1. Simply making sure we align with the existing behavior (but removing shouldn't affect anything given the comment)
2. The check is very light, it's just `(Modifiers & DeclarationModifiers.Fixed) != 0`, so we avoid the need to do more work with this small check.

Let me know if you prefer not checking it here.

Reply:
>>It is not obvious why do we still want to perform the check here under this condition?

>Which check?

Sorry for the confusion. I was referring to the ```Type.CheckAllConstraints``` that is duplicated between the two methods and performed in this method under the referenced condition.